src/lib/stores.ts

```diff
@@ -15,15 +19,68 @@ export const inSnsMenu = writable<boolean>(false);
 /* === Components === */
 export const language = writable<'en' | 'es' | 'ja' | 'ru'>('en');
 export const selectedAsset = writable<Asset>(new Asset(AssetType.ICP));
-export const selectedSns = writable<string>('BOOM DAO');
-export const snsPrincipal = writable<string>('xomae-vyaaa-aaaaq-aabhq-cai');
-export const encodedSnsIcrcAccount = writable<string>('-/-');
 export const user = writable<User | undefined>(undefined);
 export const canisters = writable<Canisters | undefined>(undefined);
 export const waterNeuronInfo = writable<WaterNeuronInfo | undefined>(undefined);
 
+/* === SNS === */
+function createSns() {
+	const { subscribe, set, update } = writable<{
+		name: string;
+		principal: string;
+		encodedAccount: string;
+		icpBalance: BigNumber | undefined;
+		nicpBalance: BigNumber | undefined;
+	}>({
+		name: '',
+		principal: '',
+		encodedAccount: '-/-',
+		icpBalance: undefined,
+		nicpBalance: undefined
```

Comment:
Isn't it weird that you define the account to be `-/-` but for balances you allow them to be undefined, and display them as `-/-`.

Reply:
It's weird but it's because encodedAccount is a string and not the others so it is possible to handle it this way. I agree that it is bad.

Reply:
wdym? It could be `string | undefined`?

Reply:
Yess I did that for consistency. But because it's a string, I can set it to -/- when it's not fetched and in my svelte component, I don't need to use {#if) ...